generic_config_updater/field_operation_validators.py

```diff
@@ -1,10 +1,116 @@
-from sonic_py_common import device_info
+import os
 import re
+import json
+import jsonpointer
+import subprocess
+from sonic_py_common import device_info
+from .gu_common import GenericConfigUpdaterError
+
+
+SCRIPT_DIR = os.path.dirname(os.path.realpath(__file__))
+GCU_TABLE_MOD_CONF_FILE = f"{SCRIPT_DIR}/gcu_field_operation_validators.conf.json"
+GET_HWSKU_CMD = "sonic-cfggen -d -v DEVICE_METADATA.localhost.hwsku"
+
+def get_asic_name():
+    asic = "unknown"
+    
+    if os.path.exists(GCU_TABLE_MOD_CONF_FILE):
+        with open(GCU_TABLE_MOD_CONF_FILE, "r") as s:
+            gcu_field_operation_conf = json.load(s)
+    else:
+        raise GenericConfigUpdaterError("GCU table modification validators config file not found")
+    
+    asic_mapping = gcu_field_operation_conf["helper_data"]["rdma_config_update_validator"]
+    asic_type = device_info.get_sonic_version_info()['asic_type'] 
+
+    if asic_type == 'cisco-8000':
+        asic = "cisco-8000"
+    elif asic_type == 'mellanox' or asic_type == 'vs' or asic_type == 'broadcom':
+        proc = subprocess.Popen(GET_HWSKU_CMD, shell=True, universal_newlines=True, stdout=subprocess.PIPE)
+        output, err = proc.communicate()
+        hwsku = output.rstrip('\n')
+        if asic_type == 'mellanox' or asic_type == 'vs':
+            spc1_hwskus = asic_mapping["mellanox_asics"]["spc1"]
+            if hwsku.lower() in [spc1_hwsku.lower() for spc1_hwsku in spc1_hwskus]:
+                asic = "spc1"
+        if asic_type == 'broadcom' or asic_type == 'vs':
+            broadcom_asics = asic_mapping["broadcom_asics"]
+            for asic_shorthand, hwskus in broadcom_asics.items():
+                if asic != "unknown":
```

Comment:
>asic [](http://example.com/codeflow?start=19&length=4)

Should you check unknown much earlier?

Reply:
asic is set to "unknown" at the start of the function. It remains unknown until one of the specified asic types in the table is identified. The purpose of this particular check is to stop iterating over all possible broadcom asics once the asic is already identified.

Reply:
If it is already identified as spc1, you can return immediately.

Reply:
Changed to return spc1 immediately